projects/packages/my-jetpack/src/class-products.php

```diff
@@ -152,6 +163,64 @@ public static function get_products() {
 		return $products;
 	}
 
+	/**
+	 * Get a list of products sorted by whether or not the user owns them
+	 * An owned product is defined as a product that is any of the following
+	 * - Active
+	 * - Has historically been active
+	 * - The user has a plan that includes the product
+	 * - The user has the standalone plugin for the product installed
+	 *
+	 * @return array
+	 */
+	public static function get_products_by_ownership() {
+		$historically_active_modules = \Jetpack_Options::get_option( 'historically_active_modules', array() );
+		$owned_active_products       = array();
+		$owned_warning_products      = array();
+		$owned_inactive_products     = array();
+		$unowned_products            = array();
+
+		foreach ( self::get_products_classes() as $class ) {
+			$product_slug    = $class::$slug;
+			$standalone_info = $class::get_standalone_info();
+			$status          = $class::get_status();
+
+			if (
+				in_array( $status, self::$active_module_statuses, true ) ||
+				$standalone_info['is_standalone_installed'] ||
+				in_array( $product_slug, $historically_active_modules, true ) ||
+				$class::has_any_plan_for_product()
+			) {
+				// This sorts the the products in the order of active -> warning -> inactive.
+				// This enables the frontend to display them in that order.
+				// This is not needed for unowned products as those will always have a status of 'inactive'
```

Comment:
This sorting was not in the design plan, it was something I came up with during development. My reasoning is that with the new way of rendering the cards, the order was already going to be different (unless we manually sort it back to what it was 🤔 ) so I figured it might be nice, in the "My Products" section, to sort them by magnitude of "activeness" of the owned product

Reply:
I like the idea, but let's maybe reach out on P2 to let @ilonagl know about changes in the plan - and ask for feedback. What do you think?